source/Engine.cpp

```diff
@@ -1248,10 +1248,22 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 	// Bail out if the ship just died.
 	if(ship->ShouldBeRemoved())
 	{
+		// Regular ships are silent about being removed.
+		if(!ship->IsSpecial())
+			return;
+		
 		// Make sure this ship's destruction was recorded, even if it died from
 		// self-destruct.
 		if(ship->IsDestroyed())
 			eventQueue.emplace_back(nullptr, ship, ShipEvent::DESTROY);
+		// The ship was able to flee. Carried ships only flee with the parent.
+		else if(!ship->CanBeCarried() || !ship->GetParent() || ship->GetParent()->ShouldBeRemoved())
```

Comment:
`!ship->GetParent()` can only be true for:
 - uninterested NPCs
 - NPCs whose flagship was destroyed since the player last took off,

since otherwise the result of `GetParent()` is the player. A `personality fleeing` ship would thus never fire this event, even it landed on a planet, since the player is still alive. In a multi-ship NPC spec, the event would fire for each non-NPC flagship, but not the NPC flagship.

Reply:
The full check is there to filters carried ships that have a live parent, the combined inverse might be clearer:
`!(ship->CanBeCarried() && ship->GetParent() && !ship->GetParent()->ShouldBeRemoved())`

Reply:
Something about "missing the forest for the trees" would be appropriate for me here :grinning: